### PR TITLE
Casino port!

### PR DESCRIPTION
**Pull request by Shadowfire117**
1571908233 shadowfire117_sergnoodle

This is first phase of casino port where I have made my casino spaceship compatible with new chomp codebase. There isnt a connecting port at station yet due to changes being made to main map, but with second phase of casino port there should be one for casino shuttle to move back and forth to transfer visitors.

Non-casino changes
Port of:
purple assistant cap, wah!
wiggler hat, dont know if the sprite works fully yet.
The monocoole, every self respecting technomancer needs this! Also need this for casino rewards.
Snaps, can recommend for lunch, I prefer the licorice flavoured ones!
Space cleaner dispenser, yep any janitors dream, so far its only on casino spaceship.
Ported old modular sofa for station usage, is already on casino, future plans could involve making the modular sofa that can be built from steel be a little more pretty and made from wood as well and also get another overlay/color when added cloth for cushion.

also deleting a .txt file that I didnt get to delete before someone really wanted me to upload wip files so they could work off them, sorry about that but its fixed now!
deleted alienwhitelist and custom_items.txt since github doesnt seem to have those blacklisted

casino port:
Ported the map with the golden goose spacecasino
ported the deluxe edition bar and kitchen gear! Say hello to the good old deluxe dispenser, deluxe food vendor and deluxe drink vendor, perhaps one day they will be constructible by crew!
ported the casino stripper pole, casino roulette, slotmachine, poker/blackjack table, ported the casino wheel of fortune
casino wallet~
casino jukebox! And re implemented the system so the casino songs no longer are on the normal jukebox, sorry folks, hope you didnt like them too much :c
casino visitor and staff manual, might need a bit of revision depending on what event managers wants
Casino chips, gotta gamble with something!
fixed runtime errors from deluxe dispenser trying to add cartridges with reagents that have been renamed on new chomp

Be aware, there will be more casino changes in near future along with some other non-casino changes, including porting of pint metamorphic glass.

### CODE REVIEW

File: code/game/machinery/casino_ch.dm [hunk 1]
```diff
@@ -0,0 +1,900 @@
...
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/milk,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/cream,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/tea,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/ice,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/mint,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/orange,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/lemon,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/lime,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/berry,
+			/*/obj/item/weapon/reagent_containers/chem_disp_cartridge/matcha_latte,*/ //Not ported yet either - Jack
```
Comment (TheFurryFeline) [inline code/game/machinery/casino_ch.dm]: Travis is screaming about this line not having a comma at the very end of it as is normal for a list. A simple addition of said missing comma at the end of this line will fix Travis's screeching.

File: code/game/machinery/casino_ch.dm [hunk 2]
```diff
@@ -0,0 +1,900 @@
...
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/apple,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/tomato,
+			/*/obj/item/weapon/reagent_containers/chem_disp_cartridge/nutbutter,*/ //Not ported yet - Jack
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/soymilk,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/grenadine,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/gingerale,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/royrogers,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/patron,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/goldschlager,
+			/*/obj/item/weapon/reagent_containers/chem_disp_cartridge/gelatin,*/ //Not ported yet - Jack
```
Comment (TheFurryFeline) [inline code/game/machinery/casino_ch.dm]: Ditto here

File: code/game/machinery/casino_ch.dm [hunk 3]
```diff
@@ -0,0 +1,900 @@
...
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/nothing,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/banana,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/honey,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/egg,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/coco,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/cherryjelly,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/carrot,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/apple,
+			/obj/item/weapon/reagent_containers/chem_disp_cartridge/tomato,
+			/*/obj/item/weapon/reagent_containers/chem_disp_cartridge/nutbutter,*/ //Not ported yet - Jack
```
Comment (TheFurryFeline) [inline code/game/machinery/casino_ch.dm]: And here.

File: code/game/machinery/casino_ch.dm [hunk 4]
```diff
@@ -0,0 +1,900 @@
...
+	products = list(/obj/item/weapon/reagent_containers/food/drinks/glass2/square = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/glass2/rocks = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/glass2/shake = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/glass2/cocktail = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/glass2/shot = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/glass2/pint = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/glass2/mug = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/glass2/wine = 25,
+					/obj/item/weapon/reagent_containers/food/drinks/metaglass = 25,
+					/*/obj/item/weapon/reagent_containers/food/drinks/metaglass/metapint = 25,*/ //Metapint glass not ported yet - Jack
```
Comment (TheFurryFeline) [inline code/game/machinery/casino_ch.dm]: Aaaaaaaand here as well.

**Comment by Darlantanis** [comment]
If you add "fixes #215" to the PR description that should close the issue automatically when this is ready to be merged.

Fixing the Travis errors TFF pointed out is preferable to avoid false positives with Travis in future PR's. Any Travis errors about active edges in z-level 11 can be ignored, these are caused by the belt miner PR. 

Please include include comments for **every line** you modified in media_tracks.dm with "CHOMPedit." This is the phrase we ctrl + F when pulling upstream so any line you edit that does not include "CHOMPedit" has a chance of being overwritten by upstream pulls and causing future headaches. 

Aside from that, this looks good.

File: code/game/machinery/casino_ch.dm (hunk 4 above)
Comment (Shadowfire117) [reply code/game/machinery/casino_ch.dm]: Done, moved the comments out to avoid issues.

**Comment by TheFurryFeline** [comment]
That works though I suggested simply adding one comma per line at the very
end of it so  it'd still count as part of the list.

**Comment by Shadowfire117** [comment]
Fixes has been made, I'll try and put in more chomp comments in future in work. Will write down to add chomp on the media tracks dm once second wave of casino stuff rolls in.

**Comment by TheFurryFeline** [comment]
There seems to be additional issues https://travis-ci.com/CHOMPStation2/CHOMPStation2/jobs/287822982#L461

**Comment by Darlantanis** [comment]
According to Travis, this doesn't compile now.

You also still need to adjust your comments on media_tracks.dm

**Comment by Shadowfire117** [comment]
But does travis do a new check when one makes edits? Cause from my understanding I fixed the issue that travis stated and it has just the old report

**Pull request merged**